### ISSUE TITLE
Compiling errors

### ISSUE THREAD

**Issue by softexpert** [carlini/c-chat-gpt-2#2]
Hello,

When compiling the code using the provided command, errors like 
these
```
/usr/bin/ld: /tmp/ccpMTKJe.o: in function `mat_isqrt':
c_chat_gpt_2.c:(.text+0x2e9): undefined reference to `sqrt'
/usr/bin/ld: /tmp/ccpMTKJe.o: in function `mat_exp':
c_chat_gpt_2.c:(.text+0x36e): undefined reference to `exp'
/usr/bin/ld: /tmp/ccpMTKJe.o: in function `tril':
c_chat_gpt_2.c:(.text+0x549): undefined reference to `exp'
/usr/bin/ld: /tmp/ccpMTKJe.o: in function `GELU':
c_chat_gpt_2.c:(.text+0x5d8): undefined reference to `tanh'
/usr/bin/ld: /tmp/ccpMTKJe.o: in function `LayerNorm':
c_chat_gpt_2.c:(.text+0x1ca7): undefined reference to `sqrt'
/usr/bin/ld: /tmp/ccpMTKJe.o: in function `main':
c_chat_gpt_2.c:(.text.startup+0x22dc): undefined reference to `sqrt'
/usr/bin/ld: c_chat_gpt_2.c:(.text.startup+0x23b1): undefined reference to `exp'
/usr/bin/ld: c_chat_gpt_2.c:(.text.startup+0x2afe): undefined reference to `tanh'
/usr/bin/ld: c_chat_gpt_2.c:(.text.startup+0x2f57): undefined reference to `sqrt'
collect2: error: ld returned 1 exit status
```
are being raised, preventing the build to finish.
However, adding `-lm` as the last parameter will resolve this (tells the compiler to link the math library).

The two provided build commands will thus become:
```
gcc -O3 c_chat_gpt_2.c -lm
```
and, respectively
```
gcc -O3 -D GOFAST c_chat_gpt_2.c -lm
```
Hope it helps ...

**Comment by abaridodo** [comment]
This change should also be applied to the run.sh file, where the compilation process takes place before inference.

**Comment by RobinHirst11** [comment]
Sounds like a skill issue tbh

**Comment by RobinHirst11** [comment]
wait no it doesn't work for me either

**Comment by carlini** [comment]
Thanks for raising this @SoftExpert. I'll fix it.

> Sounds like a skill issue tbh

famous last words. (Thanks for confirming that it's not just one isolated issue.)

**Comment by carlini** [comment]
Fixed 1d990ae8540a94e6efbe2f3071adb7d662e2447b
